Automattic/wp-calypso
# JS Logger: Use new endpoint, ES6, use tracekit

This PR continues the work from #6360 and #6481  on JS Error logger.

- Introduces Tracekit - library for normalizing stack traces
- Moves js logger to the same build flow as the rest of the application thus permitting imports and ES6
- Sends rest api data in new format used by logstash.
- Switches order of analytics parameters.  URL is now action name and error message is the label, alongside the stack trace.

## Testing
- Remove hardcoded protection in `server/pages/index.jade` (to test you need to remove the line add in line 125) :
```
			script(src=urls[ 'logger' ])
```
- break something [ call potato() in some render for example ]
- build project (after removing the line - the live reload does not work for jade)
- Put yourself in logging group: `localStorage.setItem('log-errors', 'rest-api');`
- hard reload
- Validate that you have onerror script `window.onerror`
- navigate where you broke stuff
- See that new error format is in IRC `js-errors`
- Apply DIFF `D2240` to your sandbox
- restart browser, sandbox yourself....
- navigate where you broke stuff
- Go to Kibana as described in `PCYsg-5T4-p2`
- Search `feature:calypso` on the search box
- See your error.

Screenshots atteched in arc diff D2240-code

@gwwar @lamosty @rralian
